### PR TITLE
drivers: lpadc: Make DT props match Reference manual and hardware instead of SDK enum identifiers

### PR DESCRIPTION
**Pull request by decsny**
Change the DT property value meanings for the LPADC power level and voltage reference to match what is documented for the hardware in the SOC reference manual, instead of corresponding to NXP SDK enum names.

Remove all of the build asserts in favor of devicetree enums

### CODE REVIEW

File: drivers/adc/adc_mcux_lpadc.c [hunk 1]
```diff
@@ -571,23 +550,14 @@ static const struct adc_driver_api mcux_lpadc_driver_api = {
...
 		.base = (ADC_Type *)DT_INST_REG_ADDR(n),	\
 		.clock_source = TO_LPADC_CLOCK_SOURCE(DT_INST_PROP(n, clk_source)),	\
 		.clock_div = DT_INST_PROP(n, clk_divider),					\
-		.voltage_ref =												\
-			TO_LPADC_REFERENCE_VOLTAGE(DT_INST_PROP(n, voltage_ref)),	\
-		.calibration_average =										\
-			TO_LPADC_CALIBRATION_AVERAGE(DT_INST_PROP(n, calibration_average)),	\
-		.power_level = TO_LPADC_POWER_LEVEL(DT_INST_PROP(n, power_level)),	\
+		.voltage_ref =	DT_INST_PROP(n, voltage_ref),	\
+		.calibration_average = DT_INST_ENUM_IDX(n, calibration_average),	\
```
Comment (mmahadevan108) [inline drivers/adc/adc_mcux_lpadc.c:559]: Given the `calibration_average` is not present on all SoC's (`FSL_FEATURE_LPADC_HAS_CTRL_CAL_AVGS`), please update this to support the case where this property is not specified in devicetree.
Also can you please help cleanup the below lines. We do not need the check on `FSL_FEATURE_LPADC_HAS_CTRL_CAL_AVGS`
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/adc/adc_mcux_lpadc.c#L51-L56

Comment (decsny) [reply drivers/adc/adc_mcux_lpadc.c]: sure, updated

**Comment by decsny** [comment]
@mmahadevan108 updated with your suggestion

**Review by mmahadevan108 — DISMISSED** [review]

File: drivers/adc/adc_mcux_lpadc.c [hunk 2]
```diff
@@ -571,23 +545,14 @@ static const struct adc_driver_api mcux_lpadc_driver_api = {
...
 	static const struct mcux_lpadc_config mcux_lpadc_config_##n = {	\
 		.base = (ADC_Type *)DT_INST_REG_ADDR(n),	\
 		.clock_source = TO_LPADC_CLOCK_SOURCE(DT_INST_PROP(n, clk_source)),	\
 		.clock_div = DT_INST_PROP(n, clk_divider),					\
-		.voltage_ref =												\
-			TO_LPADC_REFERENCE_VOLTAGE(DT_INST_PROP(n, voltage_ref)),	\
-		.calibration_average =										\
-			TO_LPADC_CALIBRATION_AVERAGE(DT_INST_PROP(n, calibration_average)),	\
-		.power_level = TO_LPADC_POWER_LEVEL(DT_INST_PROP(n, power_level)),	\
+		.voltage_ref =	DT_INST_PROP(n, voltage_ref),	\
```
Comment (dleach02) [inline drivers/adc/adc_mcux_lpadc.c:553]: Let's add a comment block here that highlights that the voltage_ref and calibration_average is given in units as defined in the reference manual for future users.

**Review by dleach02 — CHANGES_REQUESTED** [review]

File: drivers/adc/adc_mcux_lpadc.c (hunk 2 above)
Comment (decsny) [reply drivers/adc/adc_mcux_lpadc.c:553]: I think I will put something like this in the binding so that it will be visible in menuconfig

**Review by mmahadevan108 — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by dleach02** [comment]
@decsny please resolve the merge conflict and we can then merge tis

**Comment by carlescufi** [comment]
@decsny please rebase

**Review by dleach02 — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

**Pull request merged**